### DISCUSSION TITLE
Impossible to filter graphql data

### DISCUSSION THREAD

**Discussion by abanchev** [strapi/strapi#5875] [Legacy Topics]
Hello everyone, any inputs regarding this issue i am having?
I find it impossible to filter the fields that the graphql returns based on permissions.

I customized the controllers for my collections to make sure that only fields that the current user has the permission to "findOne" can see.
And it works, but only at the top level filtering, what i mean is this:
Let's have 3 collections A, B and C, we also have a role R that can only "find" and "findOne" A and B.

**Default situation**: (no custom controllers)
- cannot access C directly
- can access C using `{ B(id:1) { C { id } } }`

**Controllers that filter the returned data**:
- cannot access C directly
- cannot access C from B because the controller filters the fields
- can access C using `{ A(id:1) { B { C { id } } } }`

So the **controller functions do not get called** for each sub object in the path that we request.
**Neither do the policies or resolvers** for B and C if the root object is A.
And what is more the controller for A may filter some of the fields for B, however they get populated at a later time again. And the A controller never sees the data in C.

The only way to see the whole returned object is to have a middleware that can inspect the returned data. However there in the context we don't have enough data to do the filtering - no model, no user object in the context etc.

**Comment by lauriejim** [comment]
Hello, you customized your application so it will be difficult to help you considering updates we don't exactly know.
Can you also specify your Strapi version and the database you use please.

**Comment by abanchev** [comment]
Hello @lauriejim , here is a repo that has a very very simple project that shows this and one other issue

https://github.com/abanchev/strapi-issues-test

the readme has the example query that is necessary to showcase the problem.
The second query is related to this issue, if you run it you can see in the console that only the controller and policy for the top level item are executed.

**Comment by lauriejim** [comment]
I confirm this issue. 
Can you please open an issue will all this information.
